app/models/user.rb

```diff
@@ -21,6 +24,18 @@ def video_in_user_queue?(video_id)
     self.queue_items.pluck(:video_id).include?(video_id)
   end
 
+  def follows?(another_user)
+    self.following_relationships.map(&:leader).include?(another_user)
+  end
+
+  def can_follow?(another_user)
+    !(self.follows?(another_user) || self == another_user)
+  end
+
+  def full_name
+    full_name = self.first_name + " " + self.last_name
```

Comment:
There are 3 things 'wrong' here.

1. there isn't a need to store `full_name` under `full_name` local var.
2. you don't need to use `self` - and should not to use it when its not required.
3. you should interpolate rather than concatenate. `"#{first_name} #{last_name}"`